src/main/java/servlets/PlacesListServlet.java

```diff
@@ -33,12 +33,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         Double lat = Double.parseDouble(request.getParameter("lat"));
         Double lng = Double.parseDouble(request.getParameter("lng"));
         LatLng currentLocation = new LatLng(lat, lng);
+        String filters = request.getParameter("filter").replace(',', '+');
 
         NearbySearchRequest nearbySearch = new NearbySearchRequest(this.context)
             .location(currentLocation)
             .radius(500) // Metres, can change via filter
             .openNow(true)
-            .type(PlaceType.RESTAURANT);
+            .type(PlaceType.RESTAURANT)
+            .keyword(filters);
```

Comment:
When filters is an empty string (no checkbox returns), I assume that the keyword method does nothing?

Reply:
Exactly